fluss-server/src/main/java/com/alibaba/fluss/server/utils/TableDescriptorValidation.java

```diff
@@ -72,6 +78,23 @@ public static void validateTableDescriptor(TableDescriptor tableDescriptor, int
         checkMergeEngine(tableConf, hasPrimaryKey, schema);
         checkTieredLog(tableConf);
         checkPartition(tableConf, tableDescriptor.getPartitionKeys(), schema);
+        checkSystemColumns(schema);
+    }
+
+    @VisibleForTesting
+    static void checkSystemColumns(RowType schema) {
+        List<String> fieldNames = schema.getFieldNames();
+        List<String> systemColumns =
+                Arrays.asList(OFFSET_COLUMN_NAME, TIMESTAMP_COLUMN_NAME, BUCKET_COLUMN_NAME);
+        List<String> unsupportedColumns =
+                fieldNames.stream().filter(systemColumns::contains).collect(Collectors.toList());
+        if (!unsupportedColumns.isEmpty()) {
+            throw new InvalidTableException(
+                    String.format(
+                            "Unsupported columns: %s. %s are system columns, please don't use them.",
+                            String.join(", ", unsupportedColumns),
+                            String.join(", ", systemColumns)));
+        }
```

Comment:
```suggestion
            throw new InvalidTableException(
                    String.format(
                            "%s cannot be used as column names, "
                                    + "because they are reserved system columns in Fluss. "
                                    + "Please use other names for these columns. "
                                    + "The reserved system columns are: %s",
                            String.join(", ", unsupportedColumns),
                            String.join(", ", systemColumns)));
```

I improved a bit for the exception message.